addons/pos_hr/__init__.py

```diff
@@ -2,3 +2,19 @@
 
 from . import models
 from . import report
+
+from odoo import api, SUPERUSER_ID
+
+#
+# `hr_pin` module is required by `pos_hr`. But it can't depend on it
+# since `pos_hr` should auto install when only `hr` and `point_of_sale` are installed
+# (we don't want to manually install `hr_pin` to have `pos_hr` install itself).
+# This post_init_hook is a workaround to install `hr_pin` when `pos_hr` is installed.
+#
+# Note: `hr_pin` can't depend on `point_of_sale` to autoinstall because it is also
+# required by `hr_attendance` which is independant of pos.
+def _auto_install_hr_pin(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    module_list = ['hr_pin']
+    module_ids = env['ir.module.module'].search([('name', 'in', module_list), ('state', '=', 'uninstalled')])
+    module_ids.sudo().button_install()
```

Comment:
Actually won't work. The post_init_hook will install the module after the install of pos_hr.

Try to define related field on the pos.order, eg: ``barcode = fields.Char(related='employee_id.barcode')`` and it will crash:
```
2019-02-11 12:41:58,807 9977 ERROR test odoo.modules.registry: Failed to load registry 
Traceback (most recent call last):
...
  File "/home/ybrant/odoo_dev/odoo/odoo/fields.py", line 526, in _setup_related_full
    field = target._fields[name]
KeyError: 'barcode'
```

This is finally a bad idea (due to the fact that pos_hr is auto_install: True

I propose that you define the fields twice in hr_attendance and hr_pos. The ORM manages this kind of use case. Don't forget to add comments on the fields definitions to say that it exists somewhere else.

Reply:
Hmm you're right. But It would require to also define twice the views +  methods `_verify_pin` and `generate_random_barcode` (or define them in hr).
Instead I would drop this commit for now and reintroduce the module with `auto_install: ['hr_pos', 'hr_attendance']` once this [internal task](https://www.odoo.com/web?#id=1851328&action=333&active_id=1364&model=project.task&view_type=form&menu_id=4720) is done. Not a big deal if `pin` and `barcode` are defined in hr a little longer, right? What do you think?

Reply:
Good idea.